inc/widget-shortcode.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+class SiteOrigin_Panels_Widget_Shortcode {
+
+	static $text_widgets = array(
+		'SiteOrigin_Widget_Editor_Widget',
+		'SiteOrigin_Panels_Widgets_Layout',
+		'WP_Widget_Black_Studio_TinyMCE',
+		'WP_Widget_Text',
+	);
+
+	static function init() {
+		add_shortcode( 'siteorigin_widget', 'SiteOrigin_Panels_Widget_Shortcode::shortcode' );
+
+		// Integration with the cache rendering system
+		add_action( 'siteorigin_panels_start_cache_render', 'SiteOrigin_Panels_Widget_Shortcode::add_filters' );
+		add_action( 'siteorigin_panels_end_cache_render', 'SiteOrigin_Panels_Widget_Shortcode::remove_filters' );
+	}
+
+	static function add_filters() {
+		add_filter( 'siteorigin_panels_the_widget_html', 'SiteOrigin_Panels_Widget_Shortcode::widget_html', 10, 4 );
+	}
+
+	static function remove_filters(){
+		remove_filter( 'siteorigin_panels_the_widget_html', 'SiteOrigin_Panels_Widget_Shortcode::widget_html' );
+	}
+
+	/**
+	 * This shortcode just displays a widget based on the given arguments
+	 *
+	 * @param $attr
+	 * @param $content
+	 *
+	 * @return string
+	 */
+	static function shortcode( $attr, $content ){
+		$attr = shortcode_atts( array(
+			'class' => false,
+			'id' => '',
+		), $attr, 'panels_widget' );
```

Comment:
Aaah yeah. Was originally going to have different mechanisms for Widgets Bundle and Page Builder. In the end I combined them.